modules/gentemplates/openapi3-doc/src/test/java/com/reprezen/genflow/openapi3/doc/PropertyTableTest.xtend

```diff
@@ -0,0 +1,159 @@
+		                    records: 
+		                      type: array
+		                      items:
+		                        $ref: "#/components/schemas/Record"
+		                      
+		components:
+		  schemas:
+		    Record:
+		      type: object
+		      properties:
+		        id:
+		          type: integer
+		        documentDate:
+		          type: string
+		          format: date
+		        content:
+		          type: string
+	'''
+
+	private def parse(String content) {
+		val mapper = new ObjectMapper(new YAMLFactory)
+		
+		 val model = new OpenApiNormalizer(
+			ObjectType.OPENAPI3_MODEL_VERSION,
+			Option.DOC_DEFAULT_OPTIONS
+		).of(mapper.readTree(content)).normalizeToKaizen(new URL("file://test.yaml")) as OpenApi3
+
+		val generator = new XGenerateOpenApi3Doc()
+		val context = new FakeGenTemplateContext
+		context.genTargetParameters.put(OptionHelper.PREVIEW_PARAM, true)
+		generator.init(context)
+		generator.generate(model)
+	}
+
+	private def select(String doc) {
+		Jsoup.parse(doc).select(".list-group > .list-group-item:last-child > table > tbody > tr")
+	}
+
+	@Test
+	def void testResponseReturnType() {
+		val doc =Jsoup.parse(parse(model))
+		val elType = doc.select(".list-group > .list-group-item:last-child > code")
+		val elContentType = doc.select(".list-group > .list-group-item:last-child > span")
+
+		assertTrue(elType.text.contains("object[*]"))
+		assertTrue(elContentType.get(1).text.contains("application/json"))
+	}
+
+	@Test
+	def void testMethodHeader() {
+		val rows = select(parse(model))
+
+		assertEquals(10, rows.size)
+
+		val header = rows.get(0)
+		assertEquals(3, header.childNodes.size)
+		assertTrue(header.child(0).text.contains("Name"))
+		assertTrue(header.child(1).text.contains("Type"))
+		assertTrue(header.child(2).text.contains("Description"))
+	}
+
+	@Test
+	def void testMethodPrimtitiveTypes() {
+		val rows = select(parse(model))
+
+		assertEquals(10, rows.size)
+
+		val idProp = rows.get(1)
+		assertEquals(3, idProp.childNodes.size)
+		assertTrue(idProp.child(0).text.contains("id"))
+		assertTrue(idProp.child(1).text.contains("integer"))
+
+		val idPropTypeTitle = rows.get(2).select("tbody > tr > th")
+		val idPropTypeFormat = rows.get(2).select("tbody > tr > td")
+
+		assertTrue(idPropTypeTitle.text.contains("Format"))
+		assertTrue(idPropTypeFormat.text.contains("int64"))
+
+		val nameProp = rows.get(3)
+		assertEquals(3, nameProp.childNodes.size)
+		assertTrue(nameProp.child(0).text.contains("name"))
+		assertTrue(nameProp.child(1).text.contains("string"))
+		
+		val nickNamesProp = rows.get(4)
+		assertEquals(3, nickNamesProp.childNodes.size)
+		assertTrue(nickNamesProp.child(0).text.contains("nicknames"))
+		assertTrue(nickNamesProp.child(1).text.contains("string[*]"))
+
+		val recordsProp = rows.get(5)
+		assertEquals(3, recordsProp.childNodes.size)
+		assertTrue(recordsProp.child(0).text.contains("records"))
+		assertTrue(recordsProp.child(1).text.contains("Record[*]"))
```

Comment:
Something strange is happening: This test passes, as we saw in our screenshare session today. When I generate Documentation using a GenTarget, I see that it creates these named object references correctly. But the live view still shows `object[*]` instead of `Record[*]`, and there are other cases of this as well. 

I think this is because, in the live documentation view, Normalizer inlines referenced schemas, rather than localizing them. In such cases, I Normalizer injects a specification extension property (something like `x-normalizer`, also called a "vendor extension") with some trace information, including the original name. And apparently, the schema type display algorithm was successfully extracting that, in order to populate schema names in the "type" column. 

But the change in cdc87fba2a3b3206b92d44c852a9f84a3084ff01 broke that somehow, and the subsequent changes in d56a795d6419302ecb3a67061e4d1b5f901c8ccc did not fix it. 

I'll add a comment to the PR with more details on my tests. But I think we may have to back out the type naming algo change, mute the new tests (which will then be broken), and open a follow-up issue to straighten this out.

Reply:
@tedepstein I am seeing this in the live view:

![screen shot 2019-01-15 at 11 07 22](https://user-images.githubusercontent.com/148045/51173518-e1537100-18b5-11e9-847c-734ac75eeea6.png)

The live views don't use dependencies from the pom files but from the reprezen bundles, maybe you didn't update those.